### PR TITLE
refactor: update badge use plugin badges

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksfixup-legacy-badges-hashicorp.vercel.app/packer/plugins#hcp-packer-badge) 🔎
- [Asana task](https://app.asana.com/0/1203652232454021/1204196037790345) 🎟️
- [Figma](https://www.figma.com/file/v0Hyqvjdf6QJAvRkdIBV24/Dark-Mode?node-id=907-44253&t=dqYYuk1M31jSTwxa-0) 🎨 

## 🗒️ What

Fixes up the packer plugin badge to use the HDS badge component.

### Current branch
Screenshot 2023-03-28 at 9 25 54 AM
Screenshot 2023-03-28 at 9 26 32 AM

### Upstream
Screenshot 2023-03-28 at 9 26 24 AM
Screenshot 2023-03-28 at 9 26 42 AM

## 📸 Design Screenshots
Screenshot 2023-03-28 at 9 25 22 AM

## 🧪 Testing

- Visit the preview, view all the updated badges[packer/plugins](https://dev-portal-git-staging-hashicorp.vercel.app/packer/plugins#hcp-packer-badge)
- Test in both light and dark mode

## 💭 Anything else?

There's further room to remove all custom badges (not just plugin badges) from the docs content to use the HDS badge, that will be follow up work.

### CODE REVIEW

File: src/components/author-primitives/packer/plugin-badge/index.tsx [hunk 1]
```diff
@@ -18,34 +21,30 @@ type PluginLabelType =
...
 		label: 'Archived',
-		theme: 'light-gray',
-		iconSvg: false,
+		iconSvg: <IconArchive16 />,
 	},
 }
 
 function PluginBadge({ type }: { type: PluginLabelType }): React.ReactElement {
-	const { label, theme, iconSvg } = badgeTypes[type]
-	return <Badge label={label} theme={theme as BadgeTheme} iconSvg={iconSvg} />
```
Comment (kendallstrautman) [inline src/components/author-primitives/packer/plugin-badge/index.tsx:48]: TODO: need to follow up to see other use cases of this true legacy badge component.

**Review by braydenlove — APPROVED** [review]
Nice, ship it!!!

**Review by zchsh — APPROVED** [review]
✅ Code looks good! 

💭  I do wonder if we're losing a little something by dropping the color 🤔 I think it's maybe notable that [Terraform Registry](https://registry.terraform.io/providers/hashicorp/aws/latest) still uses coloured badges - I wonder whether consistency in that sense might be a competing factor to consistency with HDS?

**Comment by braydenlove** [comment]
> ✅ Code looks good!
> 
> 💭 I do wonder if we're losing a little something by dropping the color 🤔 I think it's maybe notable that [Terraform Registry](https://registry.terraform.io/providers/hashicorp/aws/latest) still uses coloured badges - I wonder whether consistency in that sense might be a competing factor to consistency with HDS?

@zchsh yeah we got this feedback on the integrations work as well. I want to holistically look at how we are handling badges across the site. For this effort I think we need dedicated time as each time we make a call we are increasing the debt. We don't have a ton of options from HDS but I do see the value in having a more robust and consistent system. My thought it that I right up a follow up ticket and we can address it then? How does that sound?

**Comment by zchsh** [comment]
@braydenlove Sounds great with me! I trust whatever direction we go, wanted to flag in case it hadn't been mentioned, but sounds like it's got lots of coverage already 💖 🙌

**Comment by braydenlove** [comment]
@zchsh when we start on it let's sync so I can get your thoughts.

**Pull request merged**